src/shared/remoteSourceTrackingService.ts

```diff
@@ -478,29 +486,49 @@ ${formatSourceMemberWarnings(outstandingSourceMembers)}`
     }
 
     // because `serverMaxRevisionCounter` is always updated, we need to select > to catch the most recent change
-    const query = `SELECT MemberType, MemberName, IsNameObsolete, RevisionCounter FROM SourceMember WHERE RevisionCounter > ${rev}`;
+    const query = `SELECT ${SOURCE_MEMBER_FIELDS.join(', ')} FROM SourceMember WHERE RevisionCounter > ${rev}`;
     this.logger[quiet ? 'silent' : 'debug'](`Query: ${query}`);
-    const queryResult = await queryFn(this.org.getConnection(), query);
-    this.queryCache.set(rev, queryResult);
-
-    return queryResult;
+    const conn = this.org.getConnection();
+    const queryResult = await queryFn(conn, query);
+    await updateCacheWithUnknownUsers(conn, queryResult, this.userQueryCache);
+    const queryResultWithResolvedUsers = queryResult.map((member) => ({
```

Comment:
converting the userId to a real name; query only those we haven't already seen and cache to avoid next time